### PR TITLE
Workflows 2.0 API Functional Tests

### PR DESCRIPTION
**Pull request by dalebremner**
@RackHD/corecommitters @srinia6 @abildw @geoff-reid

### CODE REVIEW

File: test/tests/api/v2_0/workflowTasks_tests.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from datetime import datetime
+from proboscis.asserts import * 
+from proboscis import SkipTest
+from proboscis import test
+from json import dumps, loads
+import json
+
+LOG = Log(__name__)
+
+
```
Comment (abildw) [inline test/tests/api/v2_0/workflowTasks_tests.py]: extra spacing

File: test/tests/api/v2_0/workflowTasks_tests.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        rawj=  json.loads(self.__client.last_response.data)
+        listLen =len(json.loads(self.__client.last_response.data))
+        readWorkflowTask= rawj[len(rawj)-1]
+        readFriendlyName= readWorkflowTask.get('friendlyName')
+        readInjectableName  = readWorkflowTask.get('injectableName')
+        assert_equal(readFriendlyName,self.workflowTaskDict.get('friendlyName'))
+        assert_equal(readInjectableName,self.workflowTaskDict.get('injectableName'))
+
+
+
```
Comment (abildw) [inline test/tests/api/v2_0/workflowTasks_tests.py]: extra lines

**Comment by dalebremner** [comment]
test this please

**Comment by jlongever** [comment]
@dalebremner @brianparry @geoff-reid are new client libraries required?  Has pypi been updated?

Brian should have submitted a PR for:
```
Test Name: test_node_workflows_post
Error Details: No JSON object could be decoded
```

**Comment by geoff-reid** [comment]
@jlongever I created a new client library and posted it on pypi.  @brianparry then updated the library on the rackhdci server, does he also need to up date the new jenkins slave nodes?

**Comment by jlongever** [comment]
@geoff-reid each job instance should create a new virtualenv and reinstall the libs. Shouldn't need to manually install them.

**Comment by jlongever** [comment]
@geoff-reid Is 1.0.2 sound right?

```
Downloading/unpacking on-http-api2-0>=1.0.0 (from -r build/test/requirements.txt (line 10))
  Downloading on_http_api2_0-1.0.2.tar.gz
  Running setup.py (path:/tmp/.venv/build/on-http-api2-0/setup.py) egg_info for package on-http-api2-0
```

**Comment by geoff-reid** [comment]
@jlongever Yes, 1.0.2 is the new library.

**Comment by brianparry** [comment]
https://github.com/RackHD/RackHD/pull/163 is merged, so test this please

**Comment by geoff-reid** [comment]
:+1:

File: test/tests/api/v2_0/workflows_tests.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+                readFriendlyName = readWorkflowTask.get('friendlyName')
+                readInjectableName = readWorkflowTask.get('injectableName')
+                assert_equal(readFriendlyName,self.workflowDict.get('friendlyName'))
+                assert_equal(readInjectableName,self.workflowDict.get('injectableName'))
+
+        assert_equal(foundInsertedWorkflow, True)
+
+    @test(groups=['workflows_graphs_get_name_api2'],
+          depends_on_groups=['workflows_graphs_put_api2'])
+    def test_workflows_library_id_get(self):
```
Comment (brianparry) [inline test/tests/api/v2_0/workflows_tests.py]: Should probably change the test function to something like `test_workflows_graphs_get_by_name`

File: test/tests/api/v2_0/workflows_tests.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        instance_id = rawj[0].get('id')
+        assert_is_not_none(instance_id)
+        Api().workflows_get_by_id(instance_id)
+        assert_equal(200,self.__client.last_response.status)
+
+    @test(groups=['workflows_get_id_api2'],depends_on_groups=['workflows_get_api2'])
+    def test_negative_workflows_id_get(self):
+        """ Negative Testing GET:/workflows/identifier"""
+        try:
+            Api().workflows_get_by_id("WrongIdentifier")
```
Comment (brianparry) [inline test/tests/api/v2_0/workflows_tests.py]: We should add an assertion in the try block after the Api call.  We want to make sure we get an error, so the test should fail if an exception is not raised.

**Comment by srinia6** [comment]
+1

**Comment by abildw** [comment]
@brianparry your comment was addressed with the latest commit

**Comment by brianparry** [comment]
:+1:

**Comment by brianparry** [comment]
test this please

**Comment by abildw** [comment]
test this please

**Comment by abildw** [comment]
Test this please

**Comment by brianparry** [comment]
test this please

**Comment by brianparry** [comment]
@dalebremner Tests seem to be failing in the same way consistently.

**Pull request merged**